sklearn/gaussian_process/_gpc.py

```diff
@@ -145,6 +147,16 @@ def optimizer(obj_func, initial_theta, bounds):
 
     """
 
+    _parameter_constraints = {
```

Comment:
I don't think that we should validate this class since we don't derive from it.

I think that we can directly put these validations in the `GaussianProcessClassifier` directly and avoid multiple validations.